extends Area2D
@onready var anim := $AnimatedSprite2D
@onready var player = get_tree().get_nodes_in_group("Player")[0]
var damage_done := false
var player_inside := false
var animation_playing := false  # Controla se a animaÃ§Ã£o estÃ¡ em andamento

func _process(delta: float) -> void:
	if player_inside and anim.animation == "default" and anim.frame == 3 and not damage_done:
		if player.has_method("take_damage_trap"):
			player.take_damage_trap(-10)
			damage_done = true
			print("Tomou dano")
		

func _ready() -> void:
	body_entered.connect(_on_body_entered)
	body_exited.connect(_on_body_exited)
	anim.animation_finished.connect(_on_AnimatedSprite2D_animation_finished)

func _on_body_entered(body: Node) -> void:
	if body.is_in_group("Player"): # garante que Ã© o jogador
		player_inside = true
		start_trap_animation()
		animation_playing = true

func _on_body_exited(body: Node) -> void:
	if body.is_in_group("Player"):
		player_inside = false
		animation_playing = false
		#anim.play("idle")
		# NÃ£o faz nada aqui, a animaÃ§Ã£o continuarÃ¡ atÃ© o final

func start_trap_animation() -> void:
	animation_playing = true
	damage_done = false
	anim.play("default")
	print("ğŸ’¥ Armadilha ativada!")

func _on_AnimatedSprite2D_animation_finished():
	# Verifica se a animaÃ§Ã£o que terminou foi a "default"
	if anim.animation == "default":
		#print("Executou aqui")
		animation_playing = false
		
		# Se o player ainda estiver na Ã¡rea, volta para idle (NÃƒO reinicia)
		if player_inside:
			start_trap_animation()
			print("âœ… AnimaÃ§Ã£o completada, player ainda na Ã¡rea, voltando para idle")
		else:
			# Se o player saiu, volta para idle
			anim.play("idle")
			print("ğŸ”„ Armadilha voltou ao estado inicial")
